src/App.tsx

```diff
@@ -1,28 +1,44 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { getTodos } from './api/todos';
+import { Error } from './types/Error';
+import { wait } from './utils/fetchClient';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+
+const USER_ID = 12112;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.all);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setErrorMessage('');
+
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage(Error.get))
+      .finally(() => wait(3000).then(() => setErrorMessage('')));
+  }, []);
+
+  const completedLength = todos?.filter(todo => todo.completed).length;
```

Comment:
looks like todos is alway array - no need in optional chaining
```suggestion
  const completedLength = todos.filter(todo => todo.completed).length;
```